crates/core_simd/src/vector.rs

```diff
@@ -44,6 +44,39 @@ use crate::simd::{LaneCount, Mask, MaskElement, SupportedLaneCount};
 ///
 /// [`Wrapping<T>`]: core::num::Wrapping
 ///
+/// # Layout
+/// `Simd<T, N>` has a layout similar to `[T; N]` (identical "shapes"), but with a higher alignment.
+/// `[T; N]` is aligned to `T`, but `Simd<T, N>` will have an alignment based on both `T` and `N`.
+/// It is thus sound to transmute from `Simd<T, N>` into `[T; N]` but not vice versa.
+///
+/// # ABI Bugs
+/// Due to an unfortunate bug in the Rust ABI, `Simd<T, N>` does not participate in argument-passing in SIMD registers,
+/// except as an optimization. `#[inline]` hints are recommended on functions that accept `Simd<T, N>` or return it.
+/// This may be corrected in the future.
+///
+/// # Safe SIMD with Unsafe Rust
+///
+/// `Simd<T, N>` has a higher alignment requirement than an array. It is aligned based on the size of the entire vector,
+/// not individual elements. This means it may be sound to [`transmute`] `Simd<T, N>` to `[T; N]` but **not** vice versa.
+/// When using `unsafe` Rust to read and write `Simd<T, N>` through [raw pointers], it is usually best to use
```

Comment:
I wouldn't say it is best, so much as it's something to be aware of.  I've definitely seem performance differences in some circumstances with aligned loads and stores on modern processors, not necessarily significant but there are times to use each IMO.

Reply:
If you're using slices as mentioned here, `align_to` may be both the safest and most performant option, for example.

Reply:
Mmm, I want to convey something like "try this first, rather than assuming it is automatically slower".

Reply:
I think it would be fine to say exactly that :)